common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -248,19 +248,16 @@ def _hidden_content_student_view(self, context):
         display depending on whether staff is masquerading.
         """
         if not self._can_user_view_content():
-            subsection_format = (self.format or _("subsection")).lower()  # pylint: disable=no-member
+            date_reason = "course has ended" if getattr(self, 'self_paced', False) else "due date has passed"
 
-            # Translators: subsection_format refers to the assignment
-            # type of the subsection, such as Homework, Lab, Exam, etc.
             banner_text = _(
-                "Because the due date has passed, "
-                "this {subsection_format} is hidden from the learner."
-            ).format(subsection_format=subsection_format)
+                "Because the {date_reason}, this assignment is hidden from the learner."
+            ).format(date_reason=date_reason)
 
             hidden_content_html = self.system.render_template(
                 'hidden_content.html',
                 {
-                    'subsection_format': subsection_format,
+                    'self_paced': getattr(self, 'self_paced', False),
```

Comment:
@rlucioni Thanks for pulling up the PR where this was introduced.  I still don't see the rationale for making `self_paced` an inheritable field on all xBlocks - as inheritable fields are fundamentally costlier - than just querying the value from the enveloping course.
However, I did very that the [SelfPacedDateOverrideProvider](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/self_paced_overrides.py) overrides only the date fields and not the self_paced field for blocks.  So it doesn't invalidate the earlier assumption we made in this PR - where we're getting the self_paced value only from the course module.